language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -74,26 +62,29 @@ applyTypeSchemed
     -> FoldArgs args res
     -> FoldArgsEx args
     -> [term]
-    -> m (ConstAppResult term)
+    -> m (EvaluationResult term)
 applyTypeSchemed name = go where
     go
         :: forall args'.
            TypeScheme term args' res
         -> FoldArgs args' res
         -> FoldArgsEx args'
         -> [term]
-        -> m (ConstAppResult term)
+        -> m (EvaluationResult term)
     go (TypeSchemeResult _)        y _ args =
         -- TODO: The costing function is NOT run here. Might cause problems if there's never a TypeSchemeArrow.
         case args of
-            [] -> pure . evaluationConstAppResult $ makeKnown y  -- Computed the result.
-            _  -> throwingWithCause _ConstAppError               -- Too many arguments.
-                    (ExcessArgumentsConstAppError args)
+            [] -> pure $ makeKnown y                     -- Computed the result.
+            _  -> throwingWithCause _ConstAppError       -- Too many arguments.
+                    (TooManyArgumentsConstAppError name args)
                     Nothing
     go (TypeSchemeAllType _ schK)  f exF args =
         go (schK Proxy) f exF args
     go (TypeSchemeArrow _ schB)    f exF args = case args of
-        []          -> pure ConstAppStuck                 -- Not enough arguments to compute.
+        []          ->
+            throwingWithCause _ConstAppError              -- Too few arguments.
+                (TooFewArgumentsConstAppError name)
```

Comment:
I restored the TooFewArguments error because a bug in the CK machine was giving me a message like "Too many arguments: the excess ones are []", which was a bit confusing.  I added the name of the builtin as well to give a bit more context.